test/stripe/api_test.exs

```diff
@@ -16,4 +16,34 @@ defmodule Stripe.APITest do
 
     assert Stripe.APIMock.oauth_request(:post, "www", %{body: "body"}) == :post
   end
+
+  test "oauth_request sets authorization header for deauthorize request" do
+    verify_on_exit!()
+
+    defmodule TestHackneyBehavior do
+      @typep http_success :: {:ok, integer, [{String.t(), String.t()}], String.t()}
+      @typep http_failure :: {:error, term}
+
+      @callback request(method :: atom, req_url :: String.t(), headers :: list, body :: map, opts :: list) :: http_success | http_failure
+    end
+
+    Mox.defmock(HackneyMock, for: TestHackneyBehavior)
+    Application.put_env(:stripity_stripe, :http_module, HackneyMock)
+
+    HackneyMock
+    |> expect(:request, 3, fn _, _, headers, _, _ ->
+      kv_headers = headers
+      |> Enum.reduce(%{}, fn {k, v}, acc -> Map.put(acc, k, v) end)
+      {:ok, 200, headers, Poison.encode!(kv_headers)}
```

Comment:
What if instead of `Mox` we just put_env a module with `request` that returned this?  Either way is fine with me - just wondering if `Mox` and `TestHackneyBehavior` is necessary....

Reply:
Great suggestion, thanks. I'll get it fixed.